src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs

```diff
@@ -307,13 +301,6 @@ public void TopLevelDotAccess_AttributeAmbiguity2A()
                             N(SyntaxKind.IdentifierToken, "B");
                         }
                     }
-                    M(SyntaxKind.SemicolonToken);
-                }
-            }
-            N(SyntaxKind.GlobalStatement);
-            {
-                N(SyntaxKind.EmptyStatement);
-                {
                     N(SyntaxKind.SemicolonToken);
```

Comment:
nothing like thinking you're missing a semicolon, only to be immediatley followed by an unmatched semicolon that makes its own statement.

Reply:
it's nice to see missing tokens "restored" and a cleaner parse.